packages/eslint-plugin/src/rules/expand-border-shorthand/__tests__/rule.test.ts

```diff
@@ -0,0 +1,295 @@
+import { outdent } from 'outdent';
+
+import { tester } from '../../../test-utils';
+import { expandBorderShorthand } from '../index';
+
+const packages_calls_and_imports = [
+  ['css', 'css', '@atlaskit/css'],
+  ['css', 'css', '@compiled/react'],
+  ['styled', 'styled.div', '@compiled/react'],
+  ['styled', 'styled.ul', '@compiled/react'],
+  ['cssMap', 'cssMap', '@atlaskit/css'],
+  ['cssMap', 'cssMap', '@compiled/react'],
+];
+
+tester.run('expand-border-shorthand', expandBorderShorthand, {
+  valid: [
+    ...packages_calls_and_imports.map(([pkg, call, imp]) => ({
+      name: `border shorthand less than 3 elements (${call}, ${imp})`,
+      code: outdent`
+        import {${pkg}} from '${imp}';
+
+        const styles1 = ${call}({
+          border: '1px',
+        });
+        const styles2 = ${call}({
+          border: '2px solid',
+        });
+        const styles3 = ${call}({
+          border: '3px red',
+        });
+        const styles4 = ${call}({
+          border: 'red',
+        });
+      `,
+    })),
+    ...packages_calls_and_imports.map(([pkg, call, imp]) => ({
+      name: `not using border shorthand            (${call}, ${imp})`,
+      code: outdent`
+        import {${pkg}} from '${imp}';
+
+        const styles1 = ${call}({
+          borderTop: '2px solid green',
+          borderRight: '3px dashed orange',
+          borderBottom: '4px double purple',
+          borderLeft: '5px groove teal',
+        });
+
+        const styles2 = ${call}({
+          borderWidth: '1px',
+          borderStyle: 'solid',
+          borderColor: 'black',
+        });
+      `,
+    })),
+    ...packages_calls_and_imports.map(([pkg, call, imp]) => ({
+      name: `invalid formats                       (${call}, ${imp})`,
+      code: outdent`
+          import {${pkg}} from '${imp}';
+
+          const styles1 = ${call}({
+            border: '1.5cm dotted ab',
+          });
+
+          const styles2 = ${call}({
+            border: '1.5cm dotted abcdefghijklmnopqrstuv',
+          });
+
+          const styles3 = ${call}({
+            border: '1.5cm dotted var(--color)',
+          });
+
+          const styles4 = ${call}({
+            border: 'foo bar baz'
+          });
+
+          const styles5 = ${call}({
+            border: '   1px     solid   blue   ',
+          });
+        `,
+    })),
+  ],
+  invalid: [
+    /* borderStyle borderWidth borderColor */
+
+    ...packages_calls_and_imports.map(([pkg, call, imp]) => ({
+      name: `borderStyle:keyword, borderWidth:px,      borderColor:rgb     - single property             (${call}, ${imp})`,
```

Comment:
spaces included in test name so it's easier to scan test names upon running
![Uploading Screenshot 2024-11-05 at 1.49.12 PM.png…]()